src/main/java/org/opensearch/ad/ratelimit/EntityColdStartWorker.java

```diff
@@ -114,15 +120,36 @@ protected void executeRequest(EntityRequest coldStartRequest, ActionListener<Voi
             0
         );
 
-        ActionListener<Void> failureListener = ActionListener.delegateResponse(listener, (delegateListener, e) -> {
-            if (ExceptionUtil.isOverloaded(e)) {
-                LOG.error("OpenSearch is overloaded");
-                setCoolDownStart();
+        ActionListener<Void> coldStartListener = ActionListener.wrap(r -> {
+            nodeStateManager.getAnomalyDetector(detectorId, ActionListener.wrap(detectorOptional -> {
+                try {
+                    if (!detectorOptional.isPresent()) {
+                        LOG.error(new ParameterizedMessage("fail to get detector [{}]", detectorId));
```

Comment:
Can we add a debug log here that the detector failed to get added to cache after cold start?

Reply:
The error is that we cannot find detector config from anomaly-detectors index, not failing to add it to cache after cold start. Is the error log enough?

Reply:
Make sense. Changed to 

```
if (!detectorOptional.isPresent()) {
                        LOG.error(new ParameterizedMessage("fail to load trained model [{}] to cache due to the detector not being found.", modelState.getModelId()));
                        return;
                    }
```